analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.Net.cs

```diff
@@ -0,0 +1,93 @@
+            Response.Headers.Add("Access-Control-Allow-Origin", "*"); // Noncompliant
+            Response.Headers.Add("Access-Control-Allow-Origin", Star); // Noncompliant
+            Response.Headers.Add("Access-Control-Allow-Origin", "https://trustedwebsite.com");
+            Response.Headers.Add("Access-Control-Allow-Origin", new [] {"https://trustedwebsite.com", "*"}); // Noncompliant
+            Response.Headers.Add("OtherString", "*");
+            Response.Headers.Add(HeaderNames.Age, "*");
+            Response.Headers.Add(HeaderNames.AccessControlAllowOrigin, "*"); // Noncompliant
+            Response.Headers.Add(HeaderNames.AccessControlAllowOrigin, "https://trustedwebsite.com");
+            Response.Headers.Add(header, "*");
+            Response.Headers.Add(HeaderNames.AccessControlAllowOrigin, headerValue);
+
+            Response.Headers.Append("Access-Control-Allow-Origin", "*"); // Noncompliant
+            Response.Headers.Append("Access-Control-Allow-Origin", "https://trustedwebsite.com");
+
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, "*"); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, Star); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, "https://trustedwebsite.com");
+            Response.Headers.Append("OtherString", "*");
+            Response.Headers.Append(HeaderNames.Age, "*");
+
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues("*")); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues(Star)); // Noncompliant
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues("https://trustedwebsite.com"));
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, new StringValues(new [] {"*", "https://trustedwebsite.com"})); // Noncompliant
+
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, "*, https://trustedwebsite.com"); // Compliant - FN
+            Response.Headers.Append(HeaderNames.AccessControlAllowOrigin, $"{Star}, https://trustedwebsite.com"); // Compliant - FN
+        }
+    }
+
+    public class Setup
+    {
+        private const string Star = "*";
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddCors(options =>
+            {
+                options.AddDefaultPolicy(builder =>
+                {
+                    builder.WithOrigins("*"); // Noncompliant
+                    builder.WithOrigins(Star); // Noncompliant
+                });
+
+                options.AddPolicy("EnableAllPolicy", builder =>
+                {
+                    builder.WithOrigins("https://trustedwebsite.com", "*"); // Noncompliant
```

Comment:
can you also add a case with `builder.WithOrigins("*", "*", "*");` ?

(I know it doesn't make sense, but to document it's raised only once)

Reply:
and another case, which should be compliant ([source](https://docs.microsoft.com/en-us/aspnet/core/security/cors?view=aspnetcore-5.0#set-the-allowed-origins)):
```
options.AddPolicy("MyAllowSubdomainPolicy",
    builder =>
    {
        builder.WithOrigins("https://*.example.com")
            .SetIsOriginAllowedToAllowWildcardSubdomains();
    });
```